config/main.py

```diff
@@ -1801,19 +1809,25 @@ def shutdown(ctx, interface_name):
         if interface_name is None:
             ctx.fail("'interface_name' is None!")
 
-    if interface_name_is_valid(interface_name) is False:
+    intf_fs = parse_interface_in_filter(interface_name)
+    if len(intf_fs) == 1 and interface_name_is_valid(interface_name) is False:
         ctx.fail("Interface name is invalid. Please enter a valid interface name!!")
 
-    if interface_name.startswith("Ethernet"):
-        if VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
-            config_db.mod_entry("VLAN_SUB_INTERFACE", interface_name, {"admin_status": "down"})
-        else:
-            config_db.mod_entry("PORT", interface_name, {"admin_status": "down"})
-    elif interface_name.startswith("PortChannel"):
-        if VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
-            config_db.mod_entry("VLAN_SUB_INTERFACE", interface_name, {"admin_status": "down"})
-        else:
-            config_db.mod_entry("PORTCHANNEL", interface_name, {"admin_status": "down"})
+    port_dict = config_db.get_table('PORT')
+    for port_name in port_dict.keys():
+        if port_name in intf_fs:
+            config_db.mod_entry("PORT", port_name, {"admin_status": "down"})
+
+    portchannel_list = config_db.get_table("PORTCHANNEL")
+    for po_name in portchannel_list.keys():
+        if po_name in intf_fs:
+            config_db.mod_entry("PORTCHANNEL", po_name, {"admin_status": "down"})
+
+    subport_list = config_db.get_table("VLAN_SUB_INTERFACE")
+    for sp_name in subport_list.keys():
+        if sp_name in intf_fs:
+            config_db.mod_entry("VLAN_SUB_INTERFACE", po_name, {"admin_status": "down"})
```

Comment:
still `po_name`

Reply:
I saw it change. changed sp_name to subport, how about now?

Reply:
I dint mean to change `sp_name`, in the` config_db.mod_entry,`  (line 1829), it was still using `po_name`. You can use the same name as used in `startup `to be consistent.

Reply:
Had tried change to sp_name but seems did not take effect. so changed to subport, and changed to sp_name again. hoping it works this time.